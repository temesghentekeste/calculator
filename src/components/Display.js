import React from 'react';
import PropTypes from 'prop-types';

function Display({ result }) {
  return (
    <div className="displayContainer">
      <h1 data-testid="result" className="displayContainer__result">{result}</h1>
    </div>
  );
}

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
