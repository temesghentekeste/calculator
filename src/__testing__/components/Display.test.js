import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../../components/Display';

let getByTestId;
let component;

beforeEach(() => {
  component = render(<Display />);
  getByTestId = component.getByTestId;
});

it('renders the Display component', () => {
  const component = renderer.create(<Display />).toJSON();

  expect(component).toMatchSnapshot();
});

it('renders the Display component with a result', () => {
  const component = renderer.create(<Display result='100' />).toJSON();

  expect(component).toMatchSnapshot();
});
