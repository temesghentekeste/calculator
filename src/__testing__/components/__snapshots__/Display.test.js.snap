// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Display component 1`] = `
<div
  className="displayContainer"
>
  <h1
    className="displayContainer__result"
    data-testid="result"
  >
    0
  </h1>
</div>
`;

exports[`renders the Display component with a result 1`] = `
<div
  className="displayContainer"
>
  <h1
    className="displayContainer__result"
    data-testid="result"
  >
    100
  </h1>
</div>
`;
